src/runtime.rs

```diff
@@ -153,7 +149,453 @@ pub struct FsSearchBody {
+    //                 // new_infos.push(new_info);
+    //                 break;
+    //             }
+    //         }
+    //         file_paths[new_path_idx.0 as usize].path.set_index(lengths.file_info_indices + new_info_indices.len() as u32);
+    //         // file_info_indices[original_indice_idx.0 as usize].file_info_index = FileInfoIdx(lengths.file_infos + idx as u32);
+    //         new_datas.push(original_data);
+    //         let mut new_info_index = original_info_idx;
+    //         new_info_index.dir_offset_index = lengths.folder_offsets;
+    //         new_info_index.file_info_index = FileInfoIdx(lengths.file_infos + idx as u32);
+    //         new_info_indices.push(new_info_index);
+    //         // info.info_to_data_index = InfoToDataIdx(0xFFFFFF);
+
+    //     }
+
+    //     let deduplicated_load_data = DirectoryOffset {
+    //         offset: shared_load_data.offset,
+    //         size: shared_load_data.size,
+    //         decomp_size: shared_load_data.decomp_size,
+    //         file_info_start_index: lengths.file_infos,
+    //         file_info_count: shared_load_data.file_info_count,
+    //         resource_index: lengths.folder_offsets
+    //     };
+
+    //     arc.folder_offsets = Self::recreate_array(arc.folder_offsets, lengths.folder_offsets as usize, &vec![deduplicated_load_data.clone()]);
+    //     arc.file_info_indices = Self::recreate_array(arc.file_info_indices, lengths.file_info_indices as usize, &new_info_indices);
+    //     arc.file_infos = Self::recreate_array(arc.file_infos, lengths.file_infos as usize, &new_infos);
+    //     arc.file_info_to_datas = Self::recreate_array(arc.file_info_to_datas, lengths.file_info_to_datas as usize, &new_info_to_datas);
+    //     arc.file_datas = Self::recreate_array(arc.file_datas, lengths.file_datas as usize, &new_datas);
+    //     instance.table2 = Self::extend_table(instance.table2, instance.table2_len as usize, new_info_indices.len());
+    //     instance.loaded_directory_table = Self::extend_table(instance.loaded_directory_table, instance.loaded_directory_table_size as usize, 1);
+    //     instance.loaded_directory_table_size += 1;
+    //     fs.folder_offset_count_1 += 1;
+    //     fs.file_info_index_count += new_info_indices.len() as u32;
+    //     fs.file_info_sub_index_count += new_info_to_datas.len() as u32;
+    //     fs.file_info_count += new_infos.len() as u32;
+    //     fs.file_data_count += new_datas.len() as u32;
+    //     instance.table2_len += new_info_indices.len() as u32;
+
+    //     Ok(())
+    // }
+
+    pub fn unshare_mass_loading_groups<Hash: Into<Hash40> + Clone>(paths: &Vec<Hash>) -> Result<(), LookupError> {
+        use std::collections::HashMap;
+        let paths: Vec<Hash40> = paths.iter().map(|x| {
+            x.clone().into()
+        }).collect();
+        unsafe {
```

Comment:
gonna be honest, my eyes glazed over when I first saw this function. It might become more readable if any parts of it can be made into smaller functions, but your call.

Reply:
also this is a really big function to monomorphize, just a warning